content/usage/concepts/services.md

```diff
@@ -71,3 +71,26 @@ services:
   database:
     image: mysql
 ```
+
+# Detached Services
+
+If you want to start a service container from the pipeline that does not block the pipeline, you can use `detach: true`. This can be useful if you want to test a docker container you have built.
+
+```yml
+pipeline:
+  build:
+    image: docker:latest
+    entrypoint: []
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    commands:
+      - "docker build -t example/image:latest -f ./Dockerfile ./"
+
+  example_image:
+    image: example/image:latest
+    detach: true
+    
+  test: # runs parallel to example_image
```

Comment:
I feel like for documenting detached, we should use a simpler example, such as starting a mysql server in detached mode. Right now it introduces multiple concepts (detached mode and building and testing docker image) instead of focusing on the single topic of detached mode.

With that being said, I feel like there is a ton of value in having documentation for building, detaching and then testing a container. @tonglil perhaps we could benefit from a subsection of the docs that focuses solely on building, testing and publishing docker containers, since there are a few different methods. Thoughts?

Reply:
EDIT: we had this documented in a prior version of the docs, but must have slipped through the cracks when we moved documentation sites. See below.

Service containers can also be included in the pipeline section of the Yaml using the detach parameter. This should be used when explicit control over startup order is required.

```diff
pipeline:
  build:
    image: golang
    commands:
      - go get
      - go build

  database:
    image: redis
+   detach: true

  test:
    image: golang
    commands:
      - go test
```